torch/functional.py

```diff
@@ -15,6 +16,15 @@
 Tensor = torch.Tensor
 from torch import _VF
 
+# Set a global declaring that we have opt_einsum
+import importlib.util
+if importlib.util.find_spec('opt_einsum') is not None:
+    import opt_einsum  # type: ignore[import]
+    _HAS_OPT_EINSUM = True
+else:
+    _HAS_OPT_EINSUM = False
```

Comment:
`try-except` feels more Pythonic (and we do it throughout the codebase, say in torch.testing._internal)
And why create additional var if we can simply set it to `None`?
Also, please consider keeping imports private
```suggestion
# Set a global declaring that we have opt_einsum
try:
  import opt_einsum as _opt_einsum
 except ModuleNotFoundError:
    _opt_einsum = None
```

Reply:
I was basing my code on https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_utils.py#L851, which does not recommend the try-except.

Oh yes, the imports, too, should be underscored.